docs/advanced-config.md

```diff
@@ -0,0 +1,44 @@
+| SPLUNK_ACCESS_TOKEN                  | accessToken                   |                                       | The optional access token for exporting signal data directly to SignalFx API.
+| SPLUNK_MAX_ATTR_LENGTH               | maxAttrLength                 | 1200                                  | Maximum length of string attribute value in characters. Longer values are truncated.
+| SPLUNK_TRACE_RESPONSE_HEADER_ENABLED | serverTimingEnabled           | `true`                                | Enable injection of `Server-Timing` header to HTTP responses.
+| SPLUNK_LOGS_INJECTION                | logInjectionEnabled           | `false`                               | Enable injecting of trace ID, span ID and service name to log records. Please note that the corresponding logging library instrumentation needs to be installed.
+| OTEL_RESOURCE_ATTRIBUTES             |                               |                                       | Comma-separated list of resource attributes added to every reported span. <details><summary>Example</summary>`key1=val1,key2=val2`</details>
+| OTEL_TRACE_ENABLED                   |                               | `true`                                | Globally enables tracer creation and auto-instrumentation.
+
+## Additional config options
+
+The following config options can be set by passing them as arguments to `startTracing()`.
+
+- `tracerConfig`: A JS object that is merged into the default tracer config replacing any existing keys. It's passed to the tracer provider during initialization. This can be used to customize the tracer provider or tracer. Must satisfy [`TracerConfig` interface](https://github.com/open-telemetry/opentelemetry-js/blob/71ba83a0dc51118e08e3148c788b81fe711003e7/packages/opentelemetry-tracing/src/types.ts#L26)
+
+- `spanExporterFactory`: A function that accepts the options passed to the startTracing function. Returns a new instance of SpanExporter. When set, this function is used to create a new exporter and the returned exporter will be used in the pipeline.
+
+- `spanProcessorFactory`: Returns a SpanProcessor instance or an array of SpanProcessor instances. When set, this function is be used to create one or more span processors. The returned processors are added to the global tracer provider and configured to process all spans generated by any tracer provider by the global provider. The function creates a new span exporter and uses it with each processor it creates. It may call `options.spanExporterFactory(options)` to create a new exporter as configured by the user.
+
+- `propagatorFactory`: A function that returns a new instance of a TextMapPropagator. Defaults to a composite propagator comprised of W3C [Trace Context](https://www.w3.org/TR/trace-context/) and [Baggage](https://w3c.github.io/baggage/) propagators.
+
+- `instrumentations`: Can be used to enable additional instrumentation packages.
```

Comment:
@Rauno56 Why these aren't available as ENV vars? Could we make them available as ENV vars for GA or it doesn't make sense?

Reply:
It really actually does not make sense to support ENV here.

1. The option expects constructors, not strings.
2. :point_up: could be changed to **also** support strings, but in that case, it would duplicate the list of installed instrumentations in `package.json`.